sklearn/impute/tests/test_common.py

```diff
@@ -84,3 +84,18 @@ def test_imputers_add_indicator_sparse(imputer, marker):
     imputer.set_params(add_indicator=False)
     X_trans_no_indicator = imputer.fit_transform(X)
     assert_allclose_dense_sparse(X_trans[:, :-4], X_trans_no_indicator)
+
+
+@pytest.mark.parametrize("imputer", IMPUTERS)
+def test_imputation_keep_missing_features(imputer):
+    X = np.array([[1, np.nan, 2], [3, np.nan, np.nan]])
+
+    imputer = imputer.__class__(keep_missing_features=True)
```

Comment:
more conventional would be: `imputer.set_params(keep_missing_features=True)`

Reply:
from what I checked, `set_params` is an instance method
I would like to prevent problems with recycling imputer objects that learned their own `statistics_` and other inner params